core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -121,6 +121,20 @@ final case class NonEmptyList[A](head: A, tail: List[A]) {
 
     NonEmptyList(head, buf.toList)
   }
+
+  /**
+   * Reverse this `NonEmptyList`.
+   *
+   * {{{
+   * scala> import cats.data.NonEmptyList
+   * scala> val nel = NonEmptyList.of(1, 2, 3)
+   * scala> nel.reverse
+   * res0: cats.data.NonEmptyList[Int] = NonEmptyList(3, 2, 1)
+   * }}}
+   */
+  def reverse: NonEmptyList[A] =
+    NonEmptyList.fromListUnsafe(toList.reverse)
```

Comment:
is it worth implementing this safely (and possibly faster):
```scala
  def reverse: NonEmptyList[T] = {

    @annotation.tailrec
    def go(h: T, rest: List[T], acc: List[T]): NonEmptyList[T] =
      rest match {
        case Nil => NonEmptyList(h, acc)
        case h1 :: t1 => go(h1, t1, h :: acc)
      }
    go(head, tail, Nil)
  }
```
I kind of think so, actually.